main.py

```diff
@@ -105,28 +105,26 @@ def generate_csv(results):
 @user_optional
 def markers():
     logging.debug('getting markers')
-    ne_lat = float(request.values['ne_lat'])
-    ne_lng = float(request.values['ne_lng'])
-    sw_lat = float(request.values['sw_lat'])
-    sw_lng = float(request.values['sw_lng'])
-    zoom = int(request.values['zoom'])
-    start_date = datetime.date.fromtimestamp(int(request.values['start_date']))
-    end_date = datetime.date.fromtimestamp(int(request.values['end_date']))
-    fatal = int(request.values['show_fatal'])
-    severe = int(request.values['show_severe'])
-    light = int(request.values['show_light'])
-    inaccurate = int(request.values['show_inaccurate'])
-    show_markers = bool(request.values['show_markers'])
-    show_discussions = bool(request.values['show_discussions'])
+
+    kwargs = {'ne_lat': float, 'ne_lng': float, 'sw_lat': float, 'sw_lng': float, 'zoom': int, 'show_fatal': bool,
+              'show_severe': bool, 'show_light': bool, 'approx': bool, 'accurate': bool, 'show_markers': bool,
+              'show_discussions': bool, 'show_urban': int, 'show_intersection': int, 'show_lane': int, 'show_day': int,
+              'show_holiday': int, 'show_time': int, 'start_time': int, 'end_time': int, 'weather': int, 'road': int,
+              'separation': int, 'surface': int, 'acctype': int, 'controlmeasure': int, 'district': int}
+
+    for arg in kwargs:
+        tmp = kwargs[arg](request.values[arg])
+        kwargs[arg] = tmp
+        print kwargs
+    kwargs['start_date'] = datetime.date.fromtimestamp(int(request.values['start_date']))
+    kwargs['end_date'] = datetime.date.fromtimestamp(int(request.values['end_date']))
 
     logging.debug('querying markers in bounding box')
-    is_thin = (zoom < MINIMAL_ZOOM)
-    accidents = Marker.bounding_box_query(ne_lat, ne_lng, sw_lat, sw_lng,
-                                          start_date, end_date,
-                                          fatal, severe, light, inaccurate,
-                                          show_markers, is_thin, yield_per=50)
-    discussions = DiscussionMarker.bounding_box_query(ne_lat, ne_lng,
-                                                      sw_lat, sw_lng, show_discussions)
+    is_thin = (kwargs['zoom'] < MINIMAL_ZOOM)
+    accidents = Marker.bounding_box_query(is_thin, yield_per=50, **kwargs)
+
+    discussions = DiscussionMarker.bounding_box_query(kwargs['ne_lat'], kwargs['ne_lng'],
```

Comment:
I'm not sure what you mean since I still see the old diff.
In any case, I wouldn't worry about memory since it's not *such* a large dictionary. Code duplication is what you should worry about.